Readme.md

```diff
@@ -0,0 +1,211 @@
+# Arcsecond
+
+Arcsecond is a javascript [Parser Combinator](https://en.wikipedia.org/wiki/Parser_combinator) library largely inspired by Haskell's Parsec.
+
+The name is also derrived from parsec, which in astronimical terms is an ["astronomical unit subtends an angle of one arcsecond"](https://en.wikipedia.org/wiki/Parsec).
+
+## Introduction
+
+### A simple parser
+
+A __Parser Combinator__ is a parser made by composing smaller parsers. For example, to parse strings like:
+
+```
+"Weather (today): Sunny"
+"Weather (yesterday): Cloudy"
+"Weather (one week ago): Rainy"
+```
+
+We might break this down into a few different parser:
+
+- One that can identify the literal string "Weather"
+- One that can identify a known time string inside brackets
+- One that can identify a known weather type
+
+These 3 parsers can then be combined into a single parser that is able to identify all those parts correctly as a structure. The following is how this would look using arcsecond:
+
+```javascript
+const {
+  str,
+  sequenceOf,
+  choice,
+  char,
+  parse
+} = require('arcsecond');
+
+const weatherString = str ('Weather');
+
+const timeString = sequenceOf ([
+  char ('('),
+  choice ([
+    str ('today'),
+    str ('yesterday'),
+    str ('one week ago')
+  ]),
+  char (')')
+]);
+
+const weatherType = choice ([
+  str ('Sunny'),
+  str ('Cloudy'),
+  str ('Rainy')
+]);
+
+const fullParser = sequenceOf ([
+  weatherString,
+  char (' '),
+  timeString,
+  str (': '),
+  weatherType
+]);
+
+// Create a function that can take a string and run our parser
+const parseWeatherData = parse(fullParser);
+
+parseWeatherData('Weather (today): Sunny').value
+//  -> [ 'Weather', ' ', [ '(', 'today', ')' ], ': ', 'Sunny' ]
```

Comment:
How about also using the `mapTo`  on the `fullParser` so user can see values in a nice non array way?

Reply:
never mind, you already did down